core/runtime/src/main/java/io/quarkiverse/flow/recorders/WorkflowApplicationRecorder.java

```diff
@@ -95,40 +84,51 @@ private void injectEventPublishers(final ArcContainer container, final WorkflowA
         }
     }
 
-    public Supplier<WorkflowDefinition> workflowDefinitionSupplier(String flowDescriptorClassName) {
-        return () -> {
-            try {
-                final WorkflowApplication app = Arc.container().instance(WorkflowApplication.class).get();
-                final ClassLoader cl = Thread.currentThread().getContextClassLoader();
-                final Class<?> flowClass = Class.forName(flowDescriptorClassName, true, cl);
-
-                Flow flow = (Flow) Arc.container().instance(flowClass, Any.Literal.INSTANCE).get();
-                final Workflow wf = flow.descriptor();
-
-                return app.workflowDefinition(wf);
-            } catch (RuntimeException re) {
-                throw re;
-            } catch (Throwable e) {
-                throw new RuntimeException("Failed to create WorkflowDefinition for " + flowDescriptorClassName, e);
-            }
-        };
-    }
+    private void injectSecretManager(final ArcContainer container, final WorkflowApplication.Builder builder) {
+        List<SecretManager> managers = container.select(SecretManager.class, Any.Literal.INSTANCE).stream().toList();
 
-    /**
-     * Replace the static ObjectMapper from the SDK with Quarkus' bean if presented on classpath
-     */
-    public void injectQuarkusObjectMapper() {
-        InjectableInstance<ObjectMapper> mapper = Arc.container().select(ObjectMapper.class);
-        if (mapper.isResolvable()) {
-            try {
-                final Class<?> ju = Class.forName(JsonUtils.class.getName());
-                final Field fi = ju.getDeclaredField("mapper");
-                fi.setAccessible(true);
-                fi.set(null, mapper.get());
-            } catch (Exception e) {
-                throw new RuntimeException("Failed to replace JsonUtils ObjectMapper's", e);
-            }
+        if (managers.isEmpty()) {
+            LOG.info("Flow: No SecretManager bean found; skipping so SDK can fall back to MicroProfile Config.");
+            return;
         }
 
+        List<SecretManager> usable = managers.stream()
+                .filter(sm -> {
+                    if (sm instanceof CredentialsProviderSecretManager qsm) {
+                        return qsm.hasAnyProviders();
+                    }
+                    return true;
+                })
+                .toList();
+
+        if (usable.isEmpty()) {
+            LOG.info("Flow: SecretManager beans present but none usable (no CredentialsProvider available); " +
+                    "skipping so SDK can fall back to MicroProfile Config.");
+            return;
+        }
+
+        SecretManager chosen = usable.stream()
+                .min(Comparator.comparingInt(SecretManager::priority))
+                .orElseThrow();
+
+        builder.withSecretManager(chosen);
+        LOG.info("Flow: SecretManager bound: {} (priority: {})",
+                chosen.getClass().getName(), chosen.priority());
```

Comment:
Since you do not like my previous proposal, Im not sure you got the point. Here you are using using two streams (plus conversion to intermediate list) to achieve something that can be done in one shot